nvflare/fuel/utils/pipe/pipe_handler.py

```diff
@@ -144,6 +146,14 @@ def stop(self, close_pipe=True):
     def _make_event_message(topic: str, data):
         return Message.new_request(topic, data)
 
+    def wait_for_other_end(self, timeout: Optional[float] = None):
+        start_time = time.time()
+        while True:
+            if timeout and time.time() - start_time > timeout:
+                raise TimeoutError(f"The other end is not up within {timeout} seconds")
+            if self._other_end_is_up:
+                break
```

Comment:
When self._other_end_is_up is False, this will cause extremely high CPU usage. If you use this approach, you must add a time.sleep() call.

Or it's better to use threading.Event() to wait for other side to end.

Reply:
thanks let me change this